src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -1022,4 +1026,83 @@ internal override bool TypeSymbolEquals(TypeWithAnnotations type, TypeWithAnnota
             }
         }
     }
+
+    /// <summary>
+    /// Represents a sequence of <see cref="NullableAnnotation"/> to apply to a type. The sequence of
+    /// flags is specific to how the individual <see cref="TypeSymbol"/> walk their nested types.
+    /// </summary>
+    internal sealed class NullableTransformStream
+    {
+        private static readonly ObjectPool<NullableTransformStream> Pool = new ObjectPool<NullableTransformStream>(() => new NullableTransformStream(), Environment.ProcessorCount);
+
+        private ImmutableArray<byte> _transforms;
+        private int _positionOrDefault;
+
+        private bool IsDefault => _transforms.IsDefault;
+
+        /// <summary>
+        /// Returns whether or not there is data remaining in the series. When a single flag is used this
+        /// will always return false.
+        /// </summary>
+        public bool HasUnusedTransforms => !IsDefault && _positionOrDefault >= 0 && _positionOrDefault < _transforms.Length;
+        public bool HasInsufficientData => _positionOrDefault < 0;
+        public bool IsComplete => !HasUnusedTransforms && !HasInsufficientData;
+
+        private NullableTransformStream()
+        {
+
+        }
+
+        public static NullableTransformStream Create(byte defaultTransform)
```

Comment:
>Create [](start = 46, length = 6)

Here and below. Should we call this method GetInstance for consistency with other poolable types and make it more obvious that the Free method should be called once the instance is no longer needed?